src/python_testing/TC_IDM_10_1.py

```diff
@@ -0,0 +1,125 @@
+#
+#    Copyright (c) 2022 Project CHIP Authors
+#    All rights reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+
+import logging
+
+from matter_testing_support import MatterBaseTest, async_test_body, default_matter_test_main
+from mobly import asserts
+
+STANDARD_RANGE = 0x0000
+MANUFACTURER_CODE_RANGE = range(0x0001, 0xFFF0 + 1)
+TEST_VENDOR_RANGE = range(0xFFF1, 0xFFF4 + 1)
+INVALID_VENDOR_RANGE = range(0xFFF5, 0xFFFF + 1)
+
+ATTRIBUTE_ID_RANGE = range(0x0000, 0x4FFF + 1)
+GLOBAL_ATTRIBUTE_ID_RANGE = range(0xF000, 0xFFFE)
+ATTRIBUTE_INVALID_MAX = 0xFFFF
+
+COMMAND_ID_RANGE = range(0x00, 0xFF + 1)
+COMMAND_ID_INVALID_MAX = 0xFFFF
+
+ATTRIBUTE_LIST_ID = 0xFFFB
+ACCEPTED_COMMAND_LIST_ID = 0xFFF9
+GENERATED_COMMAND_LIST_ID = 0xFFF8
+
+
+def split_element_into_prefix_suffix(element):
+    return (element >> 16), (element & 0xFFFF)
+
+
+def validate_attribute_id_range(attribute_id):
+    prefix, suffix = split_element_into_prefix_suffix(attribute_id)
+
+    if prefix == STANDARD_RANGE:
+        if suffix not in ATTRIBUTE_ID_RANGE and suffix not in GLOBAL_ATTRIBUTE_ID_RANGE or suffix == ATTRIBUTE_INVALID_MAX:
```

Comment:
```suggestion
        if suffix not in ATTRIBUTE_ID_RANGE and suffix not in GLOBAL_ATTRIBUTE_ID_RANGE :
```
because being not in both those ranges would include the max value....

Reply:
Oh yes, true. This seems to have been left over (in my mind) from when the check was reversed 😄 Thanks,